runtime/src/iree/hal/command_buffer.h

```diff
@@ -904,7 +904,7 @@ IREE_API_EXPORT iree_host_size_t iree_hal_command_buffer_validation_state_size(
     iree_hal_command_buffer_mode_t mode, iree_host_size_t binding_capacity);
 
 IREE_API_EXPORT void iree_hal_command_buffer_initialize(
-    iree_hal_device_t* device, iree_hal_command_buffer_mode_t mode,
+    iree_hal_allocator_t* device_allocator, iree_hal_command_buffer_mode_t mode,
```

Comment:
Thanks for pruning this down to just the allocator, it helps avoid needing to pass a giant state object around everywhere :)

Reply:
yeah I think I added it thinking I'd need more in validation but I'm really hoping I don't - it risks creating cycles and that's never good!